core/src/main/java/com/linecorp/armeria/server/file/AbstractBlockingHttpVfs.java

```diff
@@ -75,12 +99,33 @@ public final HttpFile get(
      * @param contentEncoding the desired {@code 'content-encoding'} header value of the file.
      *                        {@code null} to omit the header.
      * @param additionalHeaders the additional HTTP headers to add to the returned {@link HttpFile}.
-     *
      * @return the {@link HttpFile} at the specified {@code path}
+     *
+     * @deprecated Use {@link #blockingGet(Executor, String, Clock, String, HttpHeaders, MediaTypeResolver)}
+     *     instead.
      */
+    @Deprecated
     protected abstract HttpFile blockingGet(Executor fileReadExecutor, String path, Clock clock,
                                             @Nullable String contentEncoding, HttpHeaders additionalHeaders);
 
+    /**
+     * Finds the file at the specified {@code path}.
+     *
+     * @param fileReadExecutor the {@link Executor} which will perform the read operations against the file
+     * @param path an absolute path that starts with {@code '/'}, whose component separator is {@code '/'}
+     * @param clock the {@link Clock} which provides the current date and time
+     * @param contentEncoding the desired {@code 'content-encoding'} header value of the file.
+     *                        {@code null} to omit the header.
+     * @param additionalHeaders the additional HTTP headers to add to the returned {@link HttpFile}.
+     * @param mediaTypeResolver the {@link MediaTypeResolver} to determined {@link MediaType}.
+     * @return the {@link HttpFile} at the specified {@code path}
+     */
+    protected HttpFile blockingGet(Executor fileReadExecutor, String path, Clock clock,
+                                            @Nullable String contentEncoding, HttpHeaders additionalHeaders,
+                                            MediaTypeResolver mediaTypeResolver) {
+        return blockingGet(fileReadExecutor, path, clock, contentEncoding, additionalHeaders);
+    }
```

Comment:
How about making this abstract?

Reply:
I'm really new to contributing to an open-source project, so if I misunderstand something, please let me know🙏. If we make the new `blockingGet` abstract then in case that users have extended `AbstractBlockingHttpVfs` and then upgrade Armeria to a version that includes this change, it'll cause a breaking change and users are forces to implement new `blockingGet`. On the other hand, delegating the new `blockingGet` to the old one will maintain the results that users get before upgrading. By the way, I'm not so sure whether the scenario that I mentioned is valid/relevant or not😅.

Reply:
> By the way, I'm not so sure whether the scenario that I mentioned is valid/relevant or not.

I think that's totally valid and actually what I recommended to do that from the first. Could you check it again, @ikhoon, please?

Reply:
Even though it is a public API, I thought it exists for polymorphism and assume no one extends the class.
https://github.com/search?q=AbstractBlockingHttpVfs&type=code
We have added new abstract methods to interfaces and abstract classes that users rarely extend it.
But I'm not strong here. The current implementation is fine to me.

Reply:
Had a chat with @ikhoon and I agreed with the idea that makes the method abstract.

Technically, We shouldn't add an abstract method to abide by semver. However, as @ikhoon investigated, there's no one who implements `AbstractBlockingHttpVfs`(I believe there's also no one in a private repo) and it will make the code much cleaner if we just make it abstract.

Reply:
Since we assume that no one implements `AbstractBlockingHttpVfs` and 2 implementations of it aren't public class, Should we remove the deprecated `blockingGet` from `AbstractBlockingHttpVfs` also its implementation in `ClassPathHttpVfs` and `FileSystemHttpVfs` as well? 😁

Reply:
>  its implementation in ClassPathHttpVfs and FileSystemHttpVfs as well?

The current implementations of `ClassPathHttpVfs` and `FileSystemHttpVfs` of this PR look very nice. 👍